### ISSUE TITLE
Feat: Add lint rule to enforce modules to have `register/registerAsync` and/or `forRoot/forRootAsync`

### ISSUE THREAD

**Issue by tolgap** [TrilonIO/eslint-plugin#28]
# Description

We should detect whether module classes that return a `DynamicModule` are named `register`, `registerAsync`, `forRoot` or `forRootAsync`.

# Acceptance Criteria

**Incorrect**:

```ts
@Module({})
export class ProductsModule {
  static create(options): DynamicModule { // ⚠️ `Dynamic module methods must be named `register/registerAsync` or `forRoot/forRootAsync`
    return {
      module: ProductsModule,
      imports: [TypeOrmModule.forFeature([Product]),
      providers: [/* ... */],
    }
  }

  static registerAsync(options): DynamicModule { // ✅ 
    return {
      module: ProductsModule,
      imports: [TypeOrmModule.forFeature([Product]),
      providers: [/* ... */],
    }
  }
}
```

**Correct**:

```ts
@Module({})
export class ProductsModule {
  static register(options): DynamicModule { // ✅ 
    return {
      module: ProductsModule,
      imports: [TypeOrmModule.forFeature([Product]),
      providers: [/* ... */],
    }
  }

  static registerAsync(options): DynamicModule { // ✅ 
    return {
      module: ProductsModule,
      imports: [TypeOrmModule.forFeature([Product]),
      providers: [/* ... */],
    }
  }
}
```
